src/Datadog.Trace/SpanImplementation.cs

```diff
@@ -15,7 +13,7 @@ namespace Datadog.Trace
     /// tracks the duration of an operation as well as associated metadata in
     /// the form of a resource name, a service name, and user defined tags.
     /// </summary>
-    public class Span : IDisposable, ISpan
+    public class SpanImplementation : Span
```

Comment:
Wouldn't there be less code changes overall if this was something like `Span : SpanBase`?
Also, is the new abstract `Span` used anywhere, or should we revert this to `Span : IDisposable, ISpan`

Reply:
The abstract Span is used everywhere the original Span was, in both `Datadog.Trace.Core` and `Datadog.Trace`
I did this partially to avoid a slew of renames and fighting with what properties/methods to expose per assembly.
I can look again, but there were problems with the interface last time I tried, I am not recalling specifically what.